### PR TITLE
EPMLSTRCMW-237: Update .gitignore file

### PR DESCRIPTION
**Pull request by iliaonishchenko**
This MR has 2 purposes:

- Update .gitignore with files which generated during the work with vscode(`.bloop`,`.metals`, `.vscode`), windows(`null` directory) and docker (`docker` directory with junk files).
- Update the documentation with information about the branches and remove line length restriction.

### CODE REVIEW

**Comment by OlegKoshelev** [comment]
@ilyaonishenko  How about to add "null" folder to git.ignore? This folder emerges at compiling project in Windows.

**Review by OlegKoshelev — APPROVED** [review]

**Comment by iliaonishchenko** [comment]
@OlegKoshelev, added.

File: docs/git-commit-message-style-guide.md [hunk 1]
```diff
@@ -43,8 +49,7 @@ Not all commits are complex enough to warrant a body, therefore it is optional a
 when a commit requires a bit of explanation and context. Use the body to explain the **what** and **why** of a commit,
 not the how.
 
-When writing a body, the blank line between the title and the body is required and you should limit the length
-of each line to no more than 72 characters.
+When writing a body, the blank line between the title.
```
Comment (myazinn) [inline docs/git-commit-message-style-guide.md]:
```suggestion
When writing a body, the blank line between the title and the body is required.
```

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,11 @@
...
 **/logs
 .env
+.bloop
+.metals
+.vscode
+project/.bloop
+project/project/*
+project/metals.sbt
+docker
+null
```
Comment (myazinn) [inline .gitignore]: Can you please add newline to not make git mad?

**Review by myazinn — CHANGES_REQUESTED** [review]

File: docs/git-commit-message-style-guide.md (hunk 1 above)
Comment (myazinn) [reply docs/git-commit-message-style-guide.md]: Please create a ticket to make regexp meet new requirements

File: .gitignore (hunk 2 above)
Comment (iliaonishchenko) [reply .gitignore]: Done.

File: docs/git-commit-message-style-guide.md (hunk 1 above)
Comment (iliaonishchenko) [reply docs/git-commit-message-style-guide.md]: Ups..

**Review by myazinn — APPROVED** [review]
Pure perfection

**Review by DruzhininPavel — APPROVED** [review]
This is the big deal! I do need it !

**Review by gominasian — APPROVED** [review]
Nice

**Pull request merged**